saleor/warehouse/management.py

```diff
@@ -376,3 +378,110 @@ def deallocate_stock_for_order(order: "Order"):
         order_line__order=order, quantity_allocated__gt=0
     ).select_for_update(of=("self",))
     allocations.update(quantity_allocated=0)
+
+
+@traced_atomic_transaction()
+def allocate_preorders(order_lines_info: Iterable["OrderLineData"], channel_slug: str):
```

Comment:
this function gets too long and there is too much happening here, can we split it into smaller logical pieces?

Reply:
I've moved code for checking constraints and creating allocation object to the separate function. Now the `allocate_preorders` has similar structure to `allocate_stocks` function.